tfjs-backend-wasm/src/cc/prelu.cc

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ===========================================================================*/
+
+#include <emscripten.h>
+#include <xnnpack.h>
+#include <cmath>
+#include <cstdio>
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "src/cc/backend.h"
+#include "src/cc/util.h"
+
+const int kBlockSize = 48;
+
+namespace {
+// The operator cache maps the weights id to the xnn_operator_t instantiated for
+// // this set of weights.
+std::map<int, xnn_operator_t> operator_cache;
+
+void delete_xnn_operator(int weights_id) { operator_cache.erase(weights_id); }
+}  // namespace
+
+namespace tfjs {
+// We use C-style API to interface with Javascript.
+extern "C" {
+
+EMSCRIPTEN_KEEPALIVE
+void prelu(int x_id, int x_size, int weights_id, int out_id) {
+  const TensorInfo x_info = backend::get_tensor_info(x_id);
+  const TensorInfo weights_info = backend::get_tensor_info(weights_id);
+  const TensorInfo out_info = backend::get_tensor_info(out_id);
+
+  const float* x_buf = x_info.buf.f32;
+  const float* weights_buf = weights_info.buf.f32;
+  float* out_buf = out_info.buf.f32;
+
+  xnn_operator_t prelu_op = nullptr;
+  if (operator_cache.count(weights_id) == 0) {
```

Comment:
Would be slightly more efficient to do `operator_cache.find`, then compare to `operator_cache.end()` to detect if the operator is in cache. With current implementation, you typically search the cache twice: once in `operator_cache.count(...)`, and another time in `operator_cache.at(...)`.